azure-core/azure/core/pipeline/transport/aiohttp.py

```diff
@@ -95,16 +94,24 @@ def _get_request_data(self, request):
             return form_data
         return request.data
 
-    async def send(self, request: HttpRequest, **config: Any) -> AsyncHttpResponse:
+    async def create_session(self):
+        if self.session:
+            return self.session
+        session = aiohttp.ClientSession(loop=self._loop)
+        await session.__aenter__()
+        return session
+
+    async def close_session(self, session, **kwargs):
```

Comment:
I have concerns about close_session not actually closing the session in some cases. It does not sit well with me...

Reply:
Do mean in terms of simply renaming the function? Or in terms of how we're handling the session end-of-life?

Reply:
Is it a programming error (a.k.a a "bug") to call open multiple times in a row? And are the calls supposed to have a matching number of close?